src/prompt.rs

```diff
@@ -230,7 +234,15 @@ impl Prompt {
         if buffer.set_color(color.set_fg(Some(Color::Green))).is_err() {
             return safe_prompt();
         }
-        if write!(&mut buffer, " % ").is_err() {
+
+        let mut uid_ch = '%';
+        unsafe {
```

Comment:
This unsafe block can actually be reduced in size since the only unsafe piece is the ffi call. Rewritten to reduce unsafe block:

```rust
        let mut uid_ch = '%';
        let euid: u32;
        unsafe {
            euid = libc::geteuid();
        }
        if euid == UID_ROOT {
            uid_ch = '#';
        }
```

I prefer this way more and the best practices suggest making `unsafe` blocks as small as possible. Thoughts?

Reply:
You could also write it as:
```rust
let uid_ch = if UID_ROOT == unsafe { libc::geteuid() } {
    '#'
} else {
    '%'
};
```

Reply:
Updated. That's much cleaner, forgot about let with if expressions :P, thanks.